Source/Language/Flow/SetupSequencePhrase.cs

```diff
@@ -93,6 +95,38 @@ public ISetupSequentialResult<TResult> Returns(TResult value)
 			this.setup.AddReturns(value);
 			return this;
 		}
+		public ISetupSequentialResult<TResult> Returns(Delegate valueFunction)
```

Comment:
The parameter type `Delegate` doesn't seem right. It wouldn't make sense to pass *any* kind of delegate to `Returns`; how would Moq know what arguments to pass, and in case anything other than a `TResult` comes back, how to return it? I'd say the parameter type should be `Func<TResult>`. (When I quoted the code that you've copied almost verbatim (below), I only meant to demonstrate that you'd have to keep in mind the possibility that the passed `Func<TResult>` might be the desired result instead of the value returned by the `Func<>`. Like I said in the issue, I'm not sure this is necessary at all, perhaps try to write a test pretending to call `Returns` with a `Func` argument that should be returned as the result instead of being called.